internal/cli/args.go

```diff
@@ -48,44 +54,86 @@ tilt args -- --foo=bar frontend backend
+		return nil, err
+	}
 
-func (c *argsCmd) run(ctx context.Context, args []string) error {
-	// require --clear instead of an empty args list to ensure an experimental `tilt flags` doesn't unintentionally wipe state
-	if len(args) == 0 {
-		if !c.clear {
-			return errors.New("no args specified. If your intent is to empty the args, run `tilt args --clear`.")
+	cfg, err := getter.ToRESTConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	ctrlclient, err := client.New(cfg, client.Options{Scheme: v1alpha1.NewScheme()})
+	if err != nil {
+		return nil, err
+	}
+
+	return ctrlclient, err
+}
+
+func parseEditResult(b []byte) ([]string, error) {
+	lines := strings.Split(string(b), "\n")
+	var argsLine *string
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if len(line) == 0 || line[0] == '#' {
+			continue
 		}
-	} else {
-		if c.clear {
-			return errors.New("--clear cannot be specified with other values. either use --clear to clear the args or specify args to replace the args with a new (non-empty) value")
+		if argsLine != nil {
+			return nil, errors.New("cannot have multiple non-comment lines")
 		}
+		s := line
+		argsLine = &s
 	}
-	url := apiURL("set_tiltfile_args")
-	body := &bytes.Buffer{}
-	err := json.NewEncoder(body).Encode(args)
+	if argsLine == nil {
+		return nil, errors.New("must have exactly one non-comment line, found zero. If you want to clear the args, use `tilt args --clear`")
+	}
+	args, err := shellquote.Split(*argsLine)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing %q", string(b))
+	}
+
+	return args, nil
+}
+
+func (c *argsCmd) run(ctx context.Context, args []string) error {
+	ctrlclient, err := newClient(ctx)
 	if err != nil {
-		return errors.Wrap(err, "failed to encode args as json")
+		return err
 	}
 
-	res, err := c.post(url, "application/json", body)
+	var tf v1alpha1.Tiltfile
+	err = ctrlclient.Get(ctx, types.NamespacedName{Name: model.MainTiltfileManifestName.String()}, &tf)
 	if err != nil {
-		fmt.Println("tilt args requires a running Tilt instance")
-		return errors.Wrapf(err, "error making http request to Tilt at %s", url)
+		return err
 	}
-	defer func() {
-		_ = res.Body.Close()
-	}()
-
-	if res.StatusCode != http.StatusOK {
-		// don't print the response body for 404 since it's full of html and more noise than it's worth on the command line
-		if res.StatusCode != http.StatusNotFound {
-			_, err := io.Copy(os.Stderr, res.Body)
-			if err != nil {
-				return errors.Wrapf(err, "http request to Tilt returned non-OK status %s and writing the content of the http response failed", res.Status)
-			}
+
+	if c.clear {
+		if len(args) != 0 {
+			return errors.New("--clear cannot be specified with other values")
 		}
-		return fmt.Errorf("http request to Tilt failed: %s", res.Status)
+		args = nil
+	} else if len(args) == 0 {
+		input := fmt.Sprintf("# edit args for the running Tilt here\n%s\n", shellquote.Join(tf.Spec.Args...))
+		e := editor.NewDefaultEditor([]string{"EDITOR"})
```

Comment:
Probably unnecessary, but I know some apps support an app-specific variant, e.g. `GIT_EDITOR`, so we could have a `TILT_EDITOR`?

(Also, the word "editor" now looks fake to me after reading it so many times.)